config/validate.go

```diff
@@ -184,6 +184,15 @@ func validateDatatype(k string, v any, typ string) string {
 		if u.Scheme != "http" && u.Scheme != "https" {
 			return fmt.Sprintf("field %s (%v) must use an http or https scheme", k, v)
 		}
+	case "defaulttrue":
+		if !isString(v) {
+			return fmt.Sprintf("field %s must be of type DefaultTrue", k)
```

Comment:
This error message wouldn't be generated *unless* the field was of type DefaultTrue. This validation is checking the value portion of the configuration, so you actually want to check `if _, ok := v.(bool)` to make sure the user is specifying either true or false for the value.

I'm not sure you even need the UnmarshalText part below. But to confirm, you should try a couple of different values in a configuration and run the validator to make sure it does the right thing.

Reply:
This value comes in as a string, so I used `strconv.ParseBool` instead.

Reply:
According the docs, it does not seem that 1/0 are supported as bools yaml. I'll fix the validation to check for "true"/"false", "t"/"f".

Reply:
Its a little ugly, but I've updated it.